web-services/map-reduce/src/main/java/datawave/webservice/mr/configuration/MapReduceJobConfiguration.java

```diff
@@ -302,6 +316,40 @@ protected void prepareClasspath(String jobId, Job job, Path jobDir) throws Excep
         exportSystemProperties(jobId, job, fs, classpath);
     }
     
+    private File filterJar(File source, List<Pattern> patterns) {
+        File f = null;
+        try {
+            f = File.createTempFile(source.getName() + ".", "");
+            try (FileOutputStream fos = new FileOutputStream(f); ZipOutputStream zipOutputStream = new ZipOutputStream(fos)) {
+                try (ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(source))) {
+                    for (ZipEntry zipEntry = zipInputStream.getNextEntry(); zipEntry != null; zipEntry = zipInputStream.getNextEntry()) {
+                        boolean include = true;
+                        Iterator<Pattern> patternItr = patterns.iterator();
+                        while (patternItr.hasNext() && include == true) {
+                            Pattern p = patternItr.next();
+                            if (p.matcher(zipEntry.getName()).matches()) {
+                                include = false;
+                            }
+                        }
+                        if (include) {
```

Comment:
Not required, but you can simplify this quite a bit with streams...
```suggestion
                        final String entryName = zipEntry.getName();
                        if (patterns.stream().noneMatch(p -> p.matcher(entryName).matches())) {
```